<launch>
  <arg name="calib_file" default="$(find svo_ros)/param/calib/PS3eye.yaml"/>
  
  <!-- SVO node -->
  <node pkg="svo_ros" type="svo_node" name="svo" clear_params="true" >

    <!-- Camera topic to subscribe to -->
    <param name="cam0_topic" value="camera/image_raw" type="str" />

    <!-- Camera calibration file -->
    <param name="calib_file" value="$(arg calib_file)" />

    <!--Parameters-->
    <rosparam file="$(find svo_ros)/param/pinhole.yaml" />
    <param name="map_scale" value="1" />
    
    <!-- Loop closure detection parameters -->
    <param name="runlc" value="false" />
    <!--<rosparam file="$(find svo_ros)/param/loop_closing.yaml" />-->
    <!--<param name="voc_path" value="$(find svo_online_loopclosing)/vocabularies/" type="str" />-->

  </node>

  <!-- RVIZ + SVO GUI -->
  <node name="vis" pkg="rviz" type="rviz" args=" -d $(find svo_ros)/rviz_config.rviz" />
  <node name="svo_gui" pkg="rqt_gui" type="rqt_gui" args="-s rqt_svo.svo.Svo --args --topic svo" />
  
  
  <rosparam file="$(find data_aquisition)/param/params.yaml" />
  
  <node pkg="data_aquisition" type="usb_cam_node.py" name="usb_cam_node" output="screen">
    <param name="frame_id" value="camera" />
    <param name="camera_info_url" value="$(find data_aquisition)/param/calib/camera.yaml" />
  </node>
  
  <node pkg="data_aquisition" type="encoderData.py" name="encoderSensor" >
  </node>
  
  <node pkg="data_aquisition" type="monoData.py" name="monoSensor" >
  </node>
  
  <node pkg="data_aquisition" type="stereoData.py" name="stereoSensor" >
  </node>
  
  <node pkg="data_aquisition" type="synchronizer.py" name="dual_sync_node" >
  </node>
  
</launch>

